nntrainer/layers/layer_node.h

```diff
@@ -385,16 +385,28 @@ class LayerNode : public ml::train::Layer, public GraphNode {
 
   /**
    * @brief     read layer Weight & Bias data from file
-   * @param[in] file input file stream
+   * @param file input file stream
    */
   void read(std::ifstream &file);
 
   /**
    * @brief     save layer Weight & Bias data from file
-   * @param[in] file output file stream
+   * @param file output file stream
    */
   void save(std::ofstream &file) const;
 
+  /**
+   * @brief     Add extra variables per weight if the optimizer needs any.
+   * @param opt Shared pointer of optimizer
+   */
+  void addOptimizerVariable(std::shared_ptr<Optimizer> opt) {
+    if (LAYER_V2) {
+      init_context.addOptimizerVariable(opt);
```

Comment:
huh, this statement will have no effect. RunContext has already been created by this time (at graph initialize), and adding the optimizer variable now will not affect the weight. Either you need to do this before the graph initializes, or you must directly update the weights.

Reply:
Ahh... That's why added optimizer variable was not affect the weight.
I need to consider directly update the weights. Thanks 👍